numba/tests/test_random.py

```diff
@@ -1291,6 +1294,129 @@ def test_multinomial_3_tuple(self):
+            self.assertLessEqual(val, 1)
+        if size is None:
+            self.assertAlmostEqual(sample.sum(), 1, places=5)
+        else:
+            for totals in np.nditer(sample.sum(axis=-1)):
+                self.assertAlmostEqual(totals, 1, places=5)
+
+    def test_dirichlet_none(self):
+        """
+        Test dirichlet(alpha, size=None)
+        """
+        cfunc = jit(nopython=True)(numpy_dirichlet)
+        alpha = self.alpha
+        size = None
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # pvals as tuple
+        pvals = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with extreme probabilities
+        alpha = np.array([1, 1, 10000, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with noninteger probabilities
+        alpha = np.array([1, 1, 1.5, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        alpha = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+
+    def test_dirichlet_int(self):
+        """
+        Test dirichlet(alpha, size=int)
+        """
+        cfunc = jit(nopython=True)(numpy_dirichlet)
+        alpha = self.alpha
+        size=10
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # pvals as tuple
+        pvals = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with extreme probabilities
+        alpha = np.array([1, 1, 10000, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with noninteger probabilities
+        alpha = np.array([1, 1, 1.5, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        alpha = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+
+    def test_dirichlet_tuple(self):
+        """
+        Test dirichlet(alpha, size=tuple)
+        """
+        cfunc = jit(nopython=True)(numpy_dirichlet)
+        alpha = self.alpha
+        size=(10,10)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # pvals as tuple
+        pvals = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with extreme probabilities
+        alpha = np.array([1, 1, 10000, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        # A case with noninteger probabilities
+        alpha = np.array([1, 1, 1.5, 1], dtype=np.float64)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
+        alpha = tuple(alpha)
+        res = cfunc(alpha, size)
+        self._check_sample(alpha, size, res)
```

Comment:
Can this just be a loop in "alphas" x "sizes", i.e.
```python
sizes = (None, (10,), (10, 10))
alphas = (self.alpha, tuple(self.alpha),
          np.array([1, 1, 10000, 1], dtype=np.float64),np.array([1, 1, 1.5, 1], dtype=np.float64) )

for alpha, size in itertools.product(alphas, sizes):
    res = cfunc(alpha, size)
    self._check_sample(alpha, size, res)
```

Reply:
Done.